zerver/lib/events.py

```diff
@@ -1115,17 +1113,13 @@ def _draft_update_action(i: int) -> None:
             assert event["setting_name"] in UserProfile.property_types
         state[event["setting_name"]] = event["setting"]
     elif event["type"] == "update_global_notifications":
-        assert event["notification_name"] in UserProfile.notification_setting_types
+        assert event["notification_name"] in UserProfile.notification_settings_legacy
```

Comment:
We have not updated the code for event type `update_display_settings` to use `display_settings_legacy`.

Reply:
And as we are doing this, we would also need to make sure that the old event (`update_display_settings` and `update_global_notifications`) should be sent only if the settings belong to legacy dicts.

Reply:
oops, fixed. Changed `tornado.event_queue.accepts_event` for the second comment. Thanks!